src/gordon_gcp/plugins/publisher.py

```diff
@@ -43,14 +56,215 @@ class GDNSPublisher:
         error_channel (asyncio.Queue): a sink for
             :interface:`interfaces.IEventMessages` that were not
             processed due to problems.
+        http_client (http.AIOConnection):
+            A google http connection class
     """
     phase = 'publish'
 
-    def __init__(self, config, success_channel, error_channel):
+    def __init__(self, config, success_channel, error_channel, http_client):
         self.config = config
         self.success_channel = success_channel
         self.error_channel = error_channel
+        self.http_client = http_client
+        self.timeout = config['timeout']
+        self.managed_zone = config['managed_zone']
+        self.dns_zone = config['dns_zone']
+        self.project = config['project']
+        _api_version = config.get('version', 'v1')
+        self.base_url = HOST + BASE_CHANGES_ENDPOINT.format(
+            version=_api_version, project=self.project,
+            managedZone=self.managed_zone)
+        self._logger = logging.getLogger('')
+
+    # TODO: This will be eventually moved to GEventMessage
+    async def update_phase(self, event_msg, phase=None):
+        old_phase = event_msg.phase
+        event_msg.phase = phase or self.phase
+        msg = f'Updated phase from "{old_phase}" to "{event_msg.phase}".'
+        event_msg.append_to_history(msg, self.phase)
+
+    def _format_change(self, resource_record):
+        """Returns ResourceRecordSets to change.
+
+        Args:
+            resource_record (dict): dict contains record data
+        Returns:
+            ResourceRecordSets to pass
+                to the API to make changes.
+        """
+        return {
+                'kind': 'dns#resourceRecordSet',
+                'name': resource_record['name'],
+                'type': resource_record['type'],
+                'ttl': resource_record.get('ttl',
+                                           self.config.get('default_ttl', 300)),
```

Comment:
what do you mean on adding 'kind'?

Reply:
I mean something like `resource_record['kind'] = 'dns#resourceRecordSet' or 
```python
rr = dict(resource_record)
rr['kind'] = 'dns#resourceRecordSet'
```
if you want to avoid mutating the original resource_record.  My point is that it seems like there should be an easier way than to copy every parameter over one by one, when the resulting object is so similar to the original, but it's definitely a style thing, so if you think it makes it more confusing to read, that's a valid argument